test/index.test.ts

```diff
@@ -32,15 +32,39 @@ describe('HipThrusTS', () => {
       const req = { user: { _id: 'link', email: 'link@example.com' } };
       const hipRequestInstance = new UserAddedRequest(req);
```

Comment:
nice.  you removed the dep on that other library...
- [ ] 1. looks like all these are missing the "other direction" of __ extends __.
- [ ] 2. again, can you please test my theory that this approach is even correct?  It was just a theory that I didn't test (that you can trust "a extends b AND b extends a" to mean "a and be are IDENTICAL types, and neither one can be slightly broader".

Reply:
added other directions
i checked your theory in HTPipe attach data tests
in case we have const left = {
          attachData(context: { a: string }) {
            return { b: 4 };
          },
        };

        const rightFullyCovered = {
          attachData(context: { b: number }) {
            return { c: 4 };
          },
        };
and interface CorrectParam {
          a: string;
        }
        type CorrectReturnValue = PromiseOrSync<{ b: number; c: number }>;
if we add a one more key to correct param this one will return false instead of true
type ParamAssignableFromCorrect = Parameters<
          typeof pipedAtoBC.attachData
        >[0] extends CorrectParam
          ? true
          : false;
if we remove one param this one will return false instead of true
type ParamAssignableToCorrect = CorrectParam extends Parameters<
          typeof pipedAtoBC.attachData
        >[0]
          ? true
          : false;